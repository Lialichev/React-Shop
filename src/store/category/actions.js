export const GET_INFO = 'Get info';
export const getInfo = () => ({ type: GET_INFO });

export const SET_INFO = 'Set info';
export const setInfo = data => ({ type: SET_INFO, data });

export const CLEAN_INFO = 'Clean info';
export const cleanInfo = () => ({ type: CLEAN_INFO });
