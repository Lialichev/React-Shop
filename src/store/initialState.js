export const user = null;
export const info = null;
export const category = null;
export const categories = [];
export const products = [];
export const product = null;
export const error = '';
