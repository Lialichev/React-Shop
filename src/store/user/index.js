export * from './actions';
export * from './reducers';
export { default } from './sagas';
