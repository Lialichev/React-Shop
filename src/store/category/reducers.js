import { SET_INFO, CLEAN_INFO } from './actions';
import {
  category as categoryInit,
  info as infoInit
} from '../initialState';

export const category = (state = categoryInit, action) => {
  switch (action.type) {
    case 'default': {
      return action.data;
    }
  }

  return state;
};

export const info = (state = infoInit, action) => {
  switch (action.type) {
    case SET_INFO: {
      return action.data;
    }
    case CLEAN_INFO: {
      return infoInit;
    }
  }

  return state;
};
