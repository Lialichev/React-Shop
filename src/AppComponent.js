import {withRouter} from "react-router-dom";
import {connect} from 'react-redux';
import {ToastContainer} from 'react-toastr'

import {check, logout} from './store/user'
import {getInfo, cleanInfo} from './store/category'
import {cleanError} from './store/status'

import Header from 'components/header';
import Main from 'components/main';
import Footer from 'components/footer';
import {Pages} from './pages/Pages';

export class AppComponent extends Component {
  componentDidMount() {
    this.props.dispatch(check());
  }

  componentDidUpdate(prevProps) {
    if (!prevProps.user && this.props.user) {
      this.props.dispatch(getInfo());
    }

    if (prevProps.user && !this.props.user) {
      this.props.history.push('/');
      this.props.dispatch(cleanInfo());
    }

    if (!prevProps.error && this.props.error) {
      this.container.error(
        this.props.error,
        'Error!'
      );
      this.props.dispatch(cleanError());
    }
  }

  onLogout = () => {
    this.props.dispatch(logout());
  };

  render() {
    const {user, info} = this.props;

    return (
      <>
        <Header
          user={user}
          info={info}
          onLogout={this.onLogout}
        />
        <Main>
          <Pages user={user} info={info}/>
        </Main>
        <ToastContainer
          ref={ref => this.container = ref}
          className="toast-top-right"
        />
        <Footer/>
      </>
    );
  }
}

const mapState = state => ({
  user: state.user,
  info: state.info,
  error: state.error
});

export default withRouter(connect(mapState)(AppComponent));
