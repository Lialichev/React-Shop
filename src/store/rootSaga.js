import { all, put } from 'redux-saga/effects';

import { watchUser } from './user';
import { watchCategory } from './category';
import { watchCategories } from './categories';
import { watchProducts } from './products';
import { watchProduct } from './product';
import { watchCreateUser } from './createUser';
import { setError } from './status'

export function* rootSaga() {
  yield all([
    watchUser(),
    watchCategory(),
    watchCategories(),
    watchProducts(),
    watchProduct(),
    watchCreateUser(),
  ]);
}
