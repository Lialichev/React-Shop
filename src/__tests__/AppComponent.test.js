import { shallow } from 'enzyme'
import { AppComponent } from '../AppComponent'
import { getInfo, cleanInfo } from 'store/category'

describe('<AppComponent />', () => {
  it('should match to snapshot', () => {
    const wrapper = shallow(<AppComponent dispatch={_ => _} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render Header and Main', () => {
    const wrapper = shallow(<AppComponent dispatch={_ => _} />);
    expect(wrapper.find('Header').length).toBe(1);
    expect(wrapper.find('Main').length).toBe(1);
  });

  it('should send action getInfo() if user changed', () => {
    const fakeDispatch = jest.fn();
    const wrapper = shallow(<AppComponent dispatch={fakeDispatch} />);
    const user = {};

    wrapper.setProps({ user });

    expect(fakeDispatch).toHaveBeenCalledWith(getInfo());
  });
});
