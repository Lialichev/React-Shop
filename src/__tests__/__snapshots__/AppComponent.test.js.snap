// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppComponent /> should match to snapshot 1`] = `
<Fragment>
  <Header
    onLogout={[Function]}
  />
  <Main>
    <Pages />
  </Main>
  <ToastContainer
    className="toast-top-right"
    id="toast-container"
    newestOnTop={true}
    onClick={[Function]}
    preventDuplicates={true}
    toastMessageFactory={[Function]}
    toastType={
      Object {
        "error": "error",
        "info": "info",
        "success": "success",
        "warning": "warning",
      }
    }
  />
  <Footer />
</Fragment>
`;
